course7/homeworks/patrickh/complex/calculator.cpp

```diff
@@ -0,0 +1,73 @@
+#include <stdexcept>
+#include <iostream>
+#include "calculator.h"
+
+void Calculator::parse(const std::string& input)
+{
+    if (input == "+")
+        add();
+    else if (input == "-")
+        subtract();
+    else if (input == "*")
+        multiply();
+    else if (input == "/")
+        divide();
+    else
+        mStack.push(input);
```

Comment:
I don't think that allowing for implicit string->number conversion is a good API design.